src/EventSubscriber/LifecycleCallbackSubscriber.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PHPSharkTank\Anonymizer\EventSubscriber;
+
+use PHPSharkTank\Anonymizer\Annotation\PostAnonymize;
+use PHPSharkTank\Anonymizer\Annotation\PreAnonymize;
+use PHPSharkTank\Anonymizer\Event\PostAnonymizeEvent;
+use PHPSharkTank\Anonymizer\Event\PreAnonymizeEvent;
+use PHPSharkTank\Anonymizer\Loader\LoaderInterface;
+use PHPSharkTank\Anonymizer\Metadata\MethodMetadata;
+
+class LifecycleCallbackSubscriber
+{
+    private $loader;
+
+    public function __construct(LoaderInterface $loader)
```

Comment:
This should be contained in the GraphNavigator I think. This would avoid some round trips looking into the metadata. 
If we have an object graph with 100 Objects, this means it will make 200 cache lookups. This is quite expensive.

I would handle it like the serializer and doctrine orm repository, they handle it similarly.

Reply:
As long as we haven't had any performance issues that would be okay for me. Passing around the object graph would be an even worse variant for me.